File: z-clients/javascript/src/models/msgNamespaceGetOut.ts
// this file is @generated
import {
    type Retention,
    RetentionSerializer,
} from './retention';

export interface MsgNamespaceGetOut {
    name: string;
    retention: Retention;
    created: number;
    updated: number;
}

export const MsgNamespaceGetOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgNamespaceGetOut {
        return {
            name: object['name'],
            retention: RetentionSerializer._fromJsonObject(object['retention']),
            created: object['created'],
            updated: object['updated'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgNamespaceGetOut): any {
        return {
            'name': self.name,
            'retention': RetentionSerializer._toJsonObject(self.retention),
            'created': self.created,
            'updated': self.updated,
        };
    }
}

File: z-clients/rust/src/models/msg_namespace_get_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::retention::Retention;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgNamespaceGetOut {
    pub name: String,

    pub retention: Retention,

    pub created: u64,

    pub updated: u64,
}

impl MsgNamespaceGetOut {
    pub fn new(name: String, retention: Retention, created: u64, updated: u64) -> Self {
        Self {
            name,
            retention,
            created,
            updated,
        }
    }
}
